beacon-chain/state/state-native/v1/getters_state.go

```diff
@@ -1,115 +1,206 @@
 package v1
 
 import (
+	"fmt"
+
 	"github.com/pkg/errors"
-	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+	customtypes "github.com/prysmaticlabs/prysm/beacon-chain/state/state-native/custom-types"
 	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
 )
 
-// InnerStateUnsafe returns the pointer value of the underlying
+// ToProtoUnsafe returns the pointer value of the underlying
 // beacon state proto object, bypassing immutability. Use with care.
-func (b *BeaconState) InnerStateUnsafe() interface{} {
+func (b *BeaconState) ToProtoUnsafe() interface{} {
```

Comment:
Is this method used in any sensitive places ? The performance drop from doing all these copies is non trivial.

Reply:
It's used in a lot of places, but how to avoid these copies when the proto object does has `[][]byte` fields?

Reply:
Ok maybe add this to your tracking issue on something to be wary off. I am afraid of this blowing up somewhere due to all the extra memory being allocated.